driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -72,39 +85,51 @@ fn serialize_balances(state: &models::AccountState, orders: &[models::Order]) ->
 fn serialize_order(order: &models::Order, id: &str) -> serde_json::Value {
     json!({
         "accountID": account_id(order.account_id),
-        "sellToken": token_id(order.sell_token as usize),
-        "buyToken": token_id(order.buy_token as usize),
+        "sellToken": token_id(order.sell_token),
+        "buyToken": token_id(order.buy_token),
         "sellAmount": order.sell_amount.to_string(),
         "buyAmount": order.buy_amount.to_string(),
         "ID": id //TODO this should not be needed
     })
 }
 
+fn parse_token(key: &str) -> Result<u16, PriceFindingError> {
+    if key.starts_with("token") {
+        return key[5..].parse::<u16>().map_err(|_| {
+            PriceFindingError::new("Failed to parse token id", ErrorKind::ParseIntError)
```

Comment:
```suggestion
            PriceFindingError::new(format!("Failed to parse token id: {}", err), ErrorKind::ParseIntError)
```
where `err` is the error being mapped (this also requires `.map_err(|_| {` -> `.map_err(|err| {`).

Reply:
oh!